stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/MessageDispatcher.java

```diff
@@ -0,0 +1,214 @@
+import org.ballerinalang.jvm.JSONUtils;
+import org.ballerinalang.jvm.XMLFactory;
+import org.ballerinalang.jvm.types.AttachedFunction;
+import org.ballerinalang.jvm.types.BArrayType;
+import org.ballerinalang.jvm.types.BStructureType;
+import org.ballerinalang.jvm.types.BType;
+import org.ballerinalang.jvm.types.TypeTags;
+import org.ballerinalang.jvm.util.exceptions.BallerinaConnectorException;
+import org.ballerinalang.jvm.values.ObjectValue;
+import org.ballerinalang.jvm.values.connector.Executor;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * Handles and dispatched messages with data binding.
+ *
+ * @since 0.995
+ */
+public class MessageDispatcher {
+
+    private RabbitMQTransactionContext rabbitMQTransactionContext;
+    private Channel channel;
+    private boolean autoAck;
+    private ObjectValue service;
+
+    public MessageDispatcher(RabbitMQTransactionContext rabbitMQTransactionContext, ObjectValue service,
+                             Channel channel, boolean autoAck) {
+        this.rabbitMQTransactionContext = rabbitMQTransactionContext;
+        this.channel = channel;
+        this.autoAck = autoAck;
+        this.service = service;
+    }
+
+    /**
+     * Dispatch messages.
+     *
+     * @param message     Message content to be dispatched to the resource function.
+     * @param deliveryTag Delivery tag of the message.
+     * @param properties  Basic properties of the message.
+     */
+    public void handleDispatch(byte[] message, long deliveryTag, AMQP.BasicProperties properties) {
+        AttachedFunction[] attachedFunctions = service.getType().getAttachedFunctions();
+        AttachedFunction onMessageFunction;
+        if (RabbitMQConstants.FUNC_ON_MESSAGE.equals(attachedFunctions[0].getName())) {
+            onMessageFunction = attachedFunctions[0];
+        } else if (RabbitMQConstants.FUNC_ON_MESSAGE.equals(attachedFunctions[1].getName())) {
+            onMessageFunction = attachedFunctions[1];
+        } else {
+            return;
+        }
+        BType[] paramTypes = onMessageFunction.paramTypes;
+        int paramSize = paramTypes.length;
+        if (paramSize > 1) {
+            dispatchMessageWithDataBinding(message, deliveryTag, onMessageFunction, properties);
+        } else {
+            dispatchMessage(message, deliveryTag, properties);
+        }
+    }
+
+    /**
+     * Dispatch messages.
+     *
+     * @param message     Message content to be dispatched to the resource function.
+     * @param deliveryTag Delivery tag of the message.
+     */
+    private void dispatchMessage(byte[] message, long deliveryTag, AMQP.BasicProperties properties) {
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        try {
+            Executor.submit(service, RabbitMQConstants.FUNC_ON_MESSAGE, new RabbitMQResourceCallback(countDownLatch),
```

Comment:
Please use the other Executor.submit() API which accepts the Scheduler as the first parameter. We will be removing the API you have used as it creates a new Scheduler when submitting. Change the other Executor.submit() also

Reply:
Resolved.